svgnative/src/ports/gdiplus/GDIPlusSVGRenderer.cpp

```diff
@@ -344,7 +344,7 @@ void GDIPlusSVGRenderer::DrawPath(const Path& renderPath, const GraphicStyle& gr
         if (fillStyle.paint.type() == typeid(Color))
         {
             auto color = boost::get<Color>(fillStyle.paint);
-            color[3] *= fillStyle.fillOpacity;
+            color[3] *= fillStyle.fillOpacity * graphicStyle.opacity;
```

Comment:
Sure that we don't need to multiply all opacity values of previous states? Was thinking of something like:

```xml
<g opacity="0.5">
    <g opacity="0.5">
        <rect fill-opacity="0.5" fill="green"/>
    </g>
</g>
```

Reply:
Oh you're right. Thanks for pointing this out. Bascially, Skia and CG have mechanisms for transparency layers which automatically do this. But GDI+ and Cairo don't. So I'm thinking we can stack these values in a `std::stack`?

Reply:
@moazin Cairo does have a layer like concept that can be used for things like raster masks. For opacity and and clipping the internal context stack is good enough (save/restore logic).

For GDI+ yes, we would need to multiple the opacity values of previous save() operations. However, not sure what can be done for clipping. I actually don't know enough about GDI+ to give advices for clipping.

Reply:
@dirkschulze Have a look at the [documentation](https://docs.microsoft.com/en-us/windows/win32/api/gdiplusgraphics/nf-gdiplusgraphics-graphics-setclip(constgraphicspath_combinemode)) for `::SetClip`. It has a `CombineMode` parameter and setting it to intersect should solve the problem as far as I can see.

Reply:
I still wonder if it works with putting transforms on groups with clipping set before and after the transform. The PR still is a big improvement to the current behavior.

Reply:
@dirkschulze What do you mean by clipping set before and after the transform? Can you provide an example that I can test?